### PR TITLE
feat: viewer 3d api

### PR DESCRIPTION
**Pull request by polomani**

### CODE REVIEW

File: src/__tests__/resources/3d.mock.spec.ts [hunk 1]
```diff
@@ -36,11 +45,47 @@ describe('3D mocked', async () => {
       },
     },
   ];
+  const revisionReveal: RevealRevision3D = {
+    ...revision,
+    sceneThreedFiles: [
+      {
+        version: randomInt(),
```
Comment (ghost) [inline src/__tests__/resources/3d.mock.spec.ts:52]: Missing a `fileId` prop here

File: src/__tests__/resources/3d.mock.spec.ts [hunk 2]
```diff
@@ -36,11 +45,47 @@ describe('3D mocked', async () => {
...
+    ],
+  };
+  const nodesReveal: RevealNode3D[] = nodes.map(node3D => ({
+    ...node3D,
+    sectorId: randomInt(),
+  }));
+  const sectors: RevealSector3D[] = [
+    {
+      id: randomInt(),
+      path: 'ftps/stairway/to/heaven',
```
Comment (ghost) [inline src/__tests__/resources/3d.mock.spec.ts]: This will always be number based `0/3/7/1`

File: src/__tests__/resources/3d.mock.spec.ts [hunk 3]
```diff
@@ -36,11 +45,47 @@ describe('3D mocked', async () => {
...
+      parentId: randomInt(),
+      depth: randomInt(),
+      boundingBox: {
+        min: [0, 0, 0],
+        max: [100, 100, 100],
+      },
+      threedFiles: [
+        {
+          version: randomInt(),
```
Comment (ghost) [inline src/__tests__/resources/3d.mock.spec.ts:73]: Missing `fileId`

File: src/__tests__/testUtils.ts [hunk 4]
```diff
@@ -80,3 +80,7 @@ export function string2arrayBuffer(str: string) {
   }
   return buf;
 }
+
+export function randomInt() {
+  return Math.floor(Math.random() * 10000000000);
```
Comment (ghost) [inline src/__tests__/testUtils.ts:85]: Cool!

File: src/types/types.ts [hunk 5]
```diff
@@ -1167,6 +1167,74 @@ export interface NewApiKeyResponse extends ApiKeyObject {
...
+
+/**
+ * The file ID of the data file for this resource, with multiple versions supported.
+ * Use /3d/files/{id} to retrieve the file.
+ */
+export interface Versioned3DFile {
+  /**
+   * Version of the file format.
+   */
+  version?: number;
```
Comment (ghost) [inline src/types/types.ts]: `version` and `fileId` is required. I designed the API for it

File: src/__tests__/resources/3d.mock.spec.ts (hunk 2 above)
Comment (polomani) [reply src/__tests__/resources/3d.mock.spec.ts]: oh, ok ;(

File: src/types/types.ts (hunk 5 above)
Comment (polomani) [reply src/types/types.ts]: allrighty

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**